pyrmq/tests/test_publisher.py

```diff
@@ -118,3 +116,61 @@ def should_handle_different_ident():
             )
             publisher.publish({})
             publisher.publish({})
+
+
+def should_publish_to_the_routed_queue_as_specified_in_headers():
+    exchange_headers_name = "test_headers_exchange"
+
+    # Create and publish to first queue with enabled routing.
+    first_queue_args = {"routing.first": "first", "x-match": "all"}
+    first_publisher = Publisher(
+        exchange_name=exchange_headers_name,
+        exchange_type="headers",
+        queue_name="first_queue",
+        routing_key="first_queue",
+        queue_args=first_queue_args,
+    )
+    first_properties = {"headers": {"routing.first": "first"}}
+    first_publisher.publish({"test": "test"}, message_properties=first_properties)
+
+    # Create and publish to second queue with enabled routing.
+    second_queue_args = {"routing.second": "second", "x-match": "all"}
+    second_publisher = Publisher(
+        exchange_name=exchange_headers_name,
+        exchange_type="headers",
+        queue_name="second_queue",
+        routing_key="second_queue",
+        queue_args=second_queue_args,
+    )
+    second_properties = {"headers": {"routing.second": "second"}}
+    second_publisher.publish({"test": "test"}, message_properties=second_properties)
+
+    # Publish to the first queue just by setting the routing in the headers.
+    third_publisher = Publisher(
+        exchange_name=exchange_headers_name,
+        exchange_type="headers",
+        queue_args=first_queue_args,
+    )
+    third_publisher.publish({"test": "test"}, message_properties=first_properties)
+
+    response = {"count": 0}
+
+    def callback(data: Dict):
+        response["count"] = response["count"] + 1
+
+    consumer = Consumer(
+        exchange_name=exchange_headers_name,
+        queue_name="first_queue",
+        routing_key="first_queue",
+        callback=callback,
+    )
+    consumer.start()
+    wait_for_result(response, {"count": 2})
```

Comment:
how do we know if the last message here is not published in second queue?

Reply:
And there are no assertions in this test so it's not clear what we are testing.

Reply:
`wait_for_result` is the assertion. I'll just make another one to assert the second publisher having one published message.

Reply:
Oh okay, so it's not understandable. `wait_for_result` does not sound like it's asserting something.

Reply:
So rename? It's basically waiting for the result to be equal to passed parameter.

Reply:
Yes, rename it.